components/automate-ui/src/app/modules/infra-proxy/create-environment-modal/create-environment-modal.component.spec.ts

```diff
@@ -0,0 +1,179 @@
+
+import { CreateEnvironmentModalComponent } from './create-environment-modal.component';
+import { TelemetryService } from 'app/services/telemetry/telemetry.service';
+import { CreateEnvironmentSuccess, CreateEnvironmentFailure } from 'app/entities/environments/environment.action';
+import { HttpErrorResponse } from '@angular/common/http';
+import { HttpStatus } from 'app/types/types';
+import { EventEmitter } from '@angular/core';
+import { HttpClient, HttpHandler } from '@angular/common/http';
+
+
+class MockTelemetryService {
+  track() { }
+}
+
+describe('CreateEnvironmentModalComponent', () => {
+  let component: CreateEnvironmentModalComponent;
+  let fixture: ComponentFixture<CreateEnvironmentModalComponent>;
+
+  beforeEach( waitForAsync(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        MockComponent({ selector: 'chef-button', inputs: ['disabled'] }),
+        MockComponent({ selector: 'chef-loading-spinner' }),
+        MockComponent({ selector: 'chef-form-field' }),
+        MockComponent({ selector: 'chef-error' }),
+        MockComponent({ selector: 'chef-toolbar' }),
+        MockComponent({ selector: 'app-infra-tab-change' }),
+        MockComponent({ selector: 'app-infra-tab', inputs: ['active', 'disabled'] }),
+
+        MockComponent({ selector: 'chef-modal',
+          inputs: ['visible']
+        }),
+        CreateEnvironmentModalComponent
+      ],
+      providers: [
+        { provide: TelemetryService, useClass: MockTelemetryService },
+        HttpClient, HttpHandler
+      ],
+      imports: [
+        ReactiveFormsModule,
+        StoreModule.forRoot(ngrxReducers, { runtimeChecks })
+      ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CreateEnvironmentModalComponent);
+    component = fixture.componentInstance;
+    component.detailsFormGroup = new FormBuilder().group({
+      name: ['', null],
+      description: ['', null]
+    });
+    component.defaultAttrFormGroup = new FormBuilder().group({
+      default: ['', null]
+    });
+    component.overrideAttrFormGroup = new FormBuilder().group({
+      override: ['', null]
+    });
+    component.openEvent = new EventEmitter();
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+
+  describe('create environment', () => {
+    let store: Store<NgrxStateAtom>;
+    const environment = {
+      org_id: 'chef_manage',
+      server_id: 'test',
+      name: 'test',
+      description: 'test environment',
+      cookbook_versions: [{
+        id: 1,
+        name: 'aix',
+        version: '2.3.4',
+        operator: '<'
+      }],
+      default_attributes: {
+        test: 'test'
+      },
+      override_attributes: {
+        test: 'test'
+      }
+    };
+
+    beforeEach(() => {
+      store = TestBed.inject(Store);
+    });
+
+    it('opening create modal and reset tab to details tab', () => {
```

Comment:
I think this is a bit more clear.
```suggestion
    it('opening create modal resets default tab to details tab', () => {
```

Reply:
updated test line.